mautic/mautic
# Google Analytics UTM tags for emails and notifications

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Related user documentation PR URL | n
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/3516 https://github.com/mautic/mautic/issues/3106 https://github.com/mautic/mautic/issues/449
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
GA UTM tags support for emails and notifications (web & mobile).
Based on @alanhartless  recommendations from this thread https://github.com/mautic/mautic/issues/3106#issuecomment-267347852

Fund this feature https://mautic-fund.kuzmany.biz/campaign/1/google-analytics-utm-tags-for-channels

#### Steps to test this PR:
1. Apply PR
2. Call php app/console doctrine:schema:update -f
3. Edit email and add utm params in Advanced tab
4. Send email and see update add utm params to links
5. Same steps for notifications

Link should by for example: http://mautic.test/mautic/r/aa84839b312f4ca61e81fd524?ct=YTo1OntzOjY6InNvdXJjZSI7TjtzOjU6ImVtYWlsIjtOO3M6NDoic3RhdCI7czoxNDoieHh4eHh4eHh4eHh4eHgiO3M6NDoibGVhZCI7aTowO3M6NzoiY2hhbm5lbCI7YToxOntzOjU6ImVtYWlsIjtpOjM1O319&utm_medium=mediumtest&utm_name=nametest&utm_content=content+test&utm_source=sourcetest

![image](https://cloud.githubusercontent.com/assets/462477/25312099/aa7737a8-2810-11e7-8f7c-e27aeca23c5f.png)
